sync-core/src/test/java/com/cloudant/sync/query/IndexManagerTest.java

```diff
@@ -30,6 +36,67 @@
 public class IndexManagerTest extends AbstractIndexTestBase {
 
     @Test
+    public void correctlyMigratesToNewMetadataSchema() {
```

Comment:
This method would be much simpler if you created a "migrate" class method on IndexManager which took a database as an argument and performed the migration. Then you could just create a test SQLite database, set it up as in v1, then run the migration on it. This would allow you to run this test without needing to set up all the scaffolding like a datastore.